### ISSUE TITLE
The Status Gets Stuck at `MATLAB: Establishing Connection` on MacOS

### ISSUE THREAD

**Issue by JLP04** [mathworks/MATLAB-extension-for-vscode#3]
macOS Ventura 13.3.1
VSCode 1.77.3
Extension Version: v1.0.0
MATLAB Version: R2023 Update 1 (9.14.0.2239454) 64-bit (maci64)

When I open a .m file in VSCode, with the correct matlabroot (`/Applications/MATLAB_R2023a.app`) (I double checked that running `matlabroot` in MATLAB returns this, the extension shows `MATLAB: Establishing Connection` on the bottom bar of VSCode, and doesn't change past that, even after waiting at least 3-5 minutes.

**Comment by JLP04** [comment]
> When you run into this issue, do you already have MATLAB open on your machine?
> 
> Is the issue resolved if you close MATLAB and restart VS Code?

I've tried it both with and without MATLAB already open. I've also tried closing MATLAB and restarting VSCode. I can send logs if that would help the investigation, just let me know which ones to send.

**Comment by dklilley** [comment]
Are you using the new MATLAB Desktop Beta (triggered by installing [this](https://www.mathworks.com/matlabcentral/fileexchange/119593-new-desktop-for-matlab-beta) add-on in MATLAB)?

If so, we know there are issues with using the new desktop experience with this extension, and are working to ensure those are ironed out before the new desktop is fully released.

If not, can you share the contents of the Output tab in VS Code, with "MATLAB Language Server" selected from the drop-down?

**Comment by JLP04** [comment]
I'm not using the new MATLAB Desktop Beta. Here is the output of the Language Server (this was done without MATLAB already open) Note: The MATLAB process terminated was done manually by me because I wanted to try killing MATLAB in activity monitor to see if it would fix it: 

```
(16:31:33) matlabls: Log Directory: /var/folders/kv/nt1xs5dd5637lbxxgg4z40n00000gn/T/matlabls_48596
(16:31:34) matlabls: Launching MATLAB...
(16:34:16) matlabls: MATLAB process terminated
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: ENOTEMPTY: directory not empty, rmdir '/var/folders/kv/nt1xs5dd5637lbxxgg4z40n00000gn/T/matlablsTmp-ZRvGY2'] {
  errno: -66,
  code: 'ENOTEMPTY',
  syscall: 'rmdir',
  path: '/var/folders/kv/nt1xs5dd5637lbxxgg4z40n00000gn/T/matlablsTmp-ZRvGY2'
}
[Info  - 4:34:16 PM] Connection to server got closed. Server will restart.
true
(16:34:16) matlabls: Log Directory: /var/folders/kv/nt1xs5dd5637lbxxgg4z40n00000gn/T/matlabls_49197
(16:34:16) matlabls: Launching MATLAB...
```

**Comment by JLP04** [comment]
Let me know if you want the actual logs in the indicated log directory.

**Comment by dklilley** [comment]
I don't think we need the actual logs, the copied contents you provided should be sufficient. We are looking into this.

**Comment by dklilley** [comment]
I'm sorry for the delay in getting back to this.

@JLP04 - Are you still experiencing this issue? I would like to try to get a bit more logging/debug info, but don't want to ask you to post it to GitHub as it may contain some info about your code and machine. Can you reach out to support@mathworks.com and reference this issue?

I will follow back up on this issue once we have identified the root cause.

**Comment by JLP04** [comment]
No problem. I've sent the email.

**Comment by bmany** [comment]
I am also experiencing the same issue. Do we have any update with respect to this thread?

**Comment by JLP04** [comment]
> I am also experiencing the same issue in Ubuntu 22.04. Do we have any update with respect to this thread?

At least for MacOS, it's been found to be caused by a call to `path` in startup.m. I think a fix is in the works but the current workaround is to just comment out the call to `path`.
